src/main/java/org/pih/petl/job/IteratingJob.java

```diff
@@ -35,7 +43,10 @@ public void execute(final ExecutionContext context) throws Exception {
                     iterationVars.put(paramName, StrSubstitutor.replace(paramValue, context.getJobConfig().getParameters()));
                 }
                 JobConfig childConfig = configReader.getJobConfig(iterationVars, "jobTemplate");
-                iterationTasks.add(new JobExecutionTask(new ExecutionContext(context, childConfig)));
+                PetlJob petlJob = etlService.getPetlJob(childConfig);
+                JobExecution childExecution = new JobExecution(null, context.getJobExecution().getUuid(), childConfig.getDescription());
+                ExecutionContext iterationContext = new ExecutionContext(childExecution, childConfig);
+                iterationTasks.add(new JobExecutionTask(etlService, petlJob, iterationContext));
```

Comment:
The above ensures that nested jobs are also saved and tracked in their own rows in the petl_job_execution table, with reference to their parent job execution, and any configured description.